### PR TITLE
Blog post on patterns

### PR DESCRIPTION
**Pull request by jml**
Draft at https://jml.io/drafts/anti-patterns.html

This change is [Reviewable](https://reviewable.io/reviews/jml/code.mumak.net/2)

### CODE REVIEW

File: content/2016-08-17-patterns.rst [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+invoke it we don't need to comment our code to indicate that we are using a
+pattern because the name ``sorted_items`` makes it clear. If we choose, we can
+test it, optimize it, or perhaps even prove its correctness.
+
+If we figure out a better way of doing it, such as:
+
+.. code-block:: python
+
+   def sorted_items(d):
+       return [(k, d[k]) for k in sorted(d.keys())]
```
Comment (teh) [inline content/2016-08-17-patterns.rst:103]: or `sorted(d.items())` :) - maybe leave it as is to give commenters something easy to snipe :)

File: content/2016-08-17-patterns.rst [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+Of course, these days I'd much rather write:
+
+.. code-block:: haskell
+
+   import qualified Data.HashMap as Map
+   import Data.Maybe (fromJust)
+
+   sortedItems :: (Hashable k, Ord k) => Map.Map k v -> [(k, v)]
+   sortedItems d = [(k, fromJust k d) | k <- sort (Map.keys d)]
```
Comment (teh) [inline content/2016-08-17-patterns.rst]: or `sort (M.toList x)` :)

File: content/2016-08-17-patterns.rst (hunk 1 above)
Comment (jml) [inline content/2016-08-17-patterns.rst:103]: My thought was to preserve the exact behaviour of the Go one, which doesn't sort the values. I hadn't quite realised that the "real" Python version was so simple.

I mostly included the simple version so the Haskell version would be less daunting. Adding another, shorter snippet means I can simplify the Haskell snippet.

File: content/2016-08-17-patterns.rst (hunk 2 above)
Comment (jml) [inline content/2016-08-17-patterns.rst]: Quite! Changed.

FWIW, using `Data.Map` would give me `toAscList`, which I think takes advantage of the internal sorting of the tree map, but this is more closely parallel to the Go & Python versions.

**Pull request merged**